publictask/backend/plugin/src/main/kotlin/com/ritense/valtimo/backend/plugin/service/PublicTaskService.kt

```diff
@@ -0,0 +1,144 @@
+package com.ritense.valtimo.backend.plugin.service
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.ritense.authorization.AuthorizationContext.Companion.runWithoutAuthorization
+import com.ritense.form.domain.FormTaskOpenResultProperties
+import com.ritense.form.service.impl.DefaultFormSubmissionService
+import com.ritense.processlink.exception.ProcessLinkNotFoundException
+import com.ritense.processlink.service.ProcessLinkActivityService
+import com.ritense.valtimo.backend.plugin.domain.PublicTaskEntity
+import com.ritense.valtimo.backend.plugin.domain.PublicTaskData
+import com.ritense.valtimo.backend.plugin.htmlrenderer.service.HtmlRenderService
+import com.ritense.valtimo.backend.plugin.repository.PublicTaskRepository
+import mu.KotlinLogging
+import org.camunda.bpm.engine.RuntimeService
+import org.camunda.bpm.engine.delegate.DelegateExecution
+import org.camunda.bpm.engine.delegate.DelegateTask
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import java.time.LocalDate
+import java.util.UUID
+
+class PublicTaskService(
+    private val publicTaskRepository: PublicTaskRepository,
+    private val runtimeService: RuntimeService,
+    private val processLinkActivityService: ProcessLinkActivityService,
+    private val htmlRenderService: HtmlRenderService,
+    private val defaultFormSubmissionService: DefaultFormSubmissionService,
+) {
+
+    @Value("\${valtimo.url}")
+    private lateinit var baseUrl: String
```

Comment:
The baseUrl should also be in the constructor. And the `@Value` should be passed from the autoconfiguration